File: Data Structures & Algorithms/is-anagram/submission-0.java
class Solution {
    public boolean isAnagram(String s, String t) {
        HashMap<Character, Integer> sCharacters = countCharacters(s);
        HashMap<Character, Integer> tCharacters = countCharacters(t);

        return sCharacters.equals(tCharacters);
    }

    private HashMap<Character,Integer> countCharacters(String s) {
        HashMap<Character, Integer> charCount = new HashMap<>();

        int length = s.length();
        for(int i = 0  ; i < length ; i++) {
            char ch = s.charAt(i);
            if(charCount.containsKey(ch)) {
                charCount.put(ch, charCount.get(ch) + 1);
            } else {
                charCount.put(ch, 1);
            }
        }

        return charCount;
    }
}

File: Data Structures & Algorithms/is-anagram/submission-0.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        freq_s = self.freqMatrix(s)
        freq_t = self.freqMatrix(t)
        return freq_s == freq_t
    
    def freqMatrix(self, s: str) -> dict:
        freq_dict = {}
        for x in s:
            if x in freq_dict:
                freq_dict[x] = freq_dict[x] + 1
            else:
                freq_dict[x] = 1

        return freq_dict